src/viewer.js

```diff
@@ -41,6 +42,25 @@ class Viewer {
 
     /** @private {?Element} */
     this.iframe_ = null;
+
+    /** @private {!History} */
+    this.history_ = new History(this.unAttach.bind(this));
+  }
+
+  /**
+   * @param {!Function} hideViewer method that hides the viewer.
+   */
+  setViewerHider(hideViewer) {
+    /** @private {!Function} */
+    this.hideViewer_ = hideViewer;
+  }
+
+  /**
+   * Enables history Fragment for the init params.
+   */
+  enableHistoryFragment() {
```

Comment:
might be better to do what you suggest. I'll keep the setters for now and will probably refactor a couple times after more features are in